operator/pkg/reconciliation/constructor.go

```diff
@@ -122,6 +127,39 @@ func newEndpointsForAdditionalSeeds(dc *api.CassandraDatacenter) *corev1.Endpoin
 	return &endpoints
 }
 
+// newNodePortServiceForCassandraDatacenter creates a headless service owned by the CassandraDatacenter,
+// that preserves the client source IPs
+func newNodePortServiceForCassandraDatacenter(dc *api.CassandraDatacenter) *corev1.Service {
+	service := makeGenericHeadlessService(dc)
+	service.ObjectMeta.Name = dc.GetNodePortServiceName()
+
+	service.Spec.Type = "NodePort"
+	// Note: ClusterIp = "None" is not valid for NodePort
+	service.Spec.ClusterIP = ""
+	service.Spec.ExternalTrafficPolicy = corev1.ServiceExternalTrafficPolicyTypeLocal
+
+	cqlPort := dc.GetNodePortCqlPort()
+	broadcastPort := dc.GetNodePortBroadcastPort()
+
+	service.Spec.Ports = []corev1.ServicePort{
+		// Note: Port Names cannot be more than 15 characters
+		{
+			Name:       "broadcast",
+			Port:       int32(broadcastPort),
+			NodePort:   int32(broadcastPort),
+			TargetPort: intstr.FromInt(broadcastPort),
+		},
+		{
+			Name:       "native",
```

Comment:
@JeremiahDJordan do you think cql is a better name here? what about broadcast vs. internode?

Reply:
Broadcast is wrong. I would call it internode.  Both internode and native protocol (cql) ports have broadcast settings.